pkg/render/intrusion_detection.go

```diff
@@ -206,41 +206,46 @@ func (c *intrusionDetectionComponent) Objects() ([]client.Object, []client.Objec
 
 	// AD Related deployment only for management/standalone cluster
 	// When FIPS mode is enabled, we currently disable our python based images.
-	if !c.cfg.ManagedCluster && !operatorv1.IsFIPSModeEnabled(c.cfg.Installation.FIPSMode) {
-		// Service + Deployment + RBAC for AD API
-		objs = append(objs,
-			c.adAPIAllowTigeraPolicy(),
-			c.adAPIServiceAccount(),
-			c.adAPIAccessClusterRole(),
-			c.adAPIAccessRoleBinding(),
-		)
-
+	if !c.cfg.ManagedCluster {
 		shouldConfigureADStorage := len(c.cfg.IntrusionDetection.Spec.AnomalyDetection.StorageClassName) > 0
 
-		if !shouldConfigureADStorage {
-			objsToDelete = append(objsToDelete,
-				c.adPersistentVolumeClaim(),
-			)
-		} else if shouldConfigureADStorage && c.cfg.ShouldRenderADPVC {
+		if !operatorv1.IsFIPSModeEnabled(c.cfg.Installation.FIPSMode) {
+			// Service + Deployment + RBAC for AD API
 			objs = append(objs,
-				c.adPersistentVolumeClaim(),
+				c.adAPIAllowTigeraPolicy(),
+				c.adAPIServiceAccount(),
+				c.adAPIAccessClusterRole(),
+				c.adAPIAccessRoleBinding(),
 			)
-		}
 
-		objs = append(objs,
-			c.adAPIService(),
-			c.adAPIDeployment(shouldConfigureADStorage),
-		)
+			if !shouldConfigureADStorage {
+				objsToDelete = append(objsToDelete,
+					c.adPersistentVolumeClaim(),
+				)
+			} else if shouldConfigureADStorage && c.cfg.ShouldRenderADPVC {
+				objs = append(objs,
+					c.adPersistentVolumeClaim(),
+				)
+			}
 
-		// RBAC for AD Detector Pods
-		objs = append(objs,
-			c.adDetectorAllowTigeraPolicy(),
-			c.adDetectorServiceAccount(),
-			c.adDetectorSecret(),
-			c.adDetectorAccessRole(),
-			c.adDetectorRoleBinding(),
-		)
-		objs = append(objs, c.adDetectorPodTemplates()...)
+			objs = append(objs,
+				c.adAPIService(),
+				c.adAPIDeployment(shouldConfigureADStorage),
+			)
+
+			// RBAC for AD Detector Pods
+			objs = append(objs,
+				c.adDetectorAllowTigeraPolicy(),
+				c.adDetectorServiceAccount(),
+				c.adDetectorSecret(),
+				c.adDetectorAccessRole(),
+				c.adDetectorRoleBinding(),
+			)
+			objs = append(objs, c.adDetectorPodTemplates()...)
+		} else {
+			objsToDelete = append(objsToDelete, c.adAPIService())
+			objsToDelete = append(objsToDelete, c.adAPIDeployment(shouldConfigureADStorage))
```

Comment:
actually maybe it's better to force `shouldConfigureADStorage` to `false`  since if true it will try to grab the Storageclassname value from IDS CR

Reply:
This is addressed in b0e2ea1e0bdd0cedc01ce30496f314e1f941b2cb.